Задача 1. Положим r = 0, ti = t = 1, pi = p = 0.02.

Пусть CF1 = 50000 — единственная задолженность от одного контрагента.

R = 1000; VaR1%(X) =?; VaR5%(X) =?; ESF1%(X) =?; ESF5%(X) =?.

In [66]:
from scipy.stats import binom
from scipy.stats import poisson
p = 0.02

In [67]:
ESF1 = 50000 * p / 0.01
ESF5 = 50000 * p / 0.05
print(f"ESF1%(X)= {ESF1}, ESF5%(X)= {ESF5}")

ESF1%(X)= 100000.0, ESF5%(X)= 20000.0


Решение задачи 1:

VaR1%(X) = 50000;

VaR5%(X) = 0; 

ESF1%(X) = 50000; ограничено единственной задолженностью

ESF5%(X) = 20000.

Задача 2. Пусть CF1 = · · · = CF5 = 10000 — одинаковые задолженности от
пяти контрагентов. Пусть также дефолты контрагентов представляют
собой независимые в совокупности события. 

Найти
R =?; V aR1%(X) =?; V aR5%(X) =?; ESF1%(X) =?; ESF5%(X) =?.


In [68]:
n=5
a = []
for k in range(1, 6):
    i = binom.pmf(k,n,p)
    a.append(i)
    print(f"Вероятность наступления дефолта {k} из {n} равна {binom.pmf(k,n,p)} {sum(a)}")

Вероятность наступления дефолта 1 из 5 равна 0.09223681600000001 0.09223681600000001
Вероятность наступления дефолта 2 из 5 равна 0.0037647679999999995 0.09600158400000001
Вероятность наступления дефолта 3 из 5 равна 7.683200000000003e-05 0.09607841600000001
Вероятность наступления дефолта 4 из 5 равна 7.840000000000006e-07 0.09607920000000002
Вероятность наступления дефолта 5 из 5 равна 3.2000000000000005e-09 0.09607920320000002


In [69]:
ESF1 = 0.096 * 10000 / 0.01
ESF5 = 0.096 * 10000 / 0.05
print(f"ESF1%(X)= {ESF1}, ESF5%(X)= {ESF5}")

ESF1%(X)= 96000.0, ESF5%(X)= 19200.0


Решение задачи 2:

R = 1000;

VaR1%(X) = 10000;

VaR5%(X) = 10000; 

ESF1%(X) = 50000;

ESF5%(X) = 19200.

Задача 3. Пусть CF1 = · · · = CF50 = 1000 — одинаковые задолженности от

большого количества (50) контрагентов. Пусть также дефолты 

контрагентов представляют собой независимые в совокупности события.

R =?; V aR1%(X) =?; V aR5%(X) =?; ESF1%(X) =?; ESF5%(X) =?.

In [70]:
n=50
a = []
for k in range(1, 51):
    i = binom.pmf(k,n,p)
    a.append(i)
    print(f"Вероятность наступления дефолта {k} из {n} равна {binom.pmf(k,n,p)} {sum(a)}")

Вероятность наступления дефолта 1 из 50 равна 0.37160171437460954 0.37160171437460954
Вероятность наступления дефолта 2 из 50 равна 0.18580085718730477 0.5574025715619143
Вероятность наступления дефолта 3 из 50 равна 0.060669667652997375 0.6180722392149117
Вероятность наступления дефолта 4 из 50 равна 0.014548338671892216 0.6326205778868039
Вероятность наступления дефолта 5 из 50 равна 0.0027315248118654854 0.6353521026986694
Вероятность наступления дефолта 6 из 50 равна 0.0004180905324283901 0.6357701932310977
Вероятность наступления дефолта 7 из 50 равна 5.363260474300058e-05 0.6358238258358407
Вероятность наступления дефолта 8 из 50 равна 5.883168377420997e-06 0.6358297090042182
Вероятность наступления дефолта 9 из 50 равна 5.603017502305686e-07 0.6358302693059684
Вероятность наступления дефолта 10 из 50 равна 4.6882391345822964e-08 0.6358303161883597
Вероятность наступления дефолта 11 из 50 равна 3.479212715831026e-09 0.6358303196675724
Вероятность наступления дефолта 12 из 50 равн

In [71]:
n=50
a = []
for k in range(1, 51):
    i = binom.pmf(k,n,p)
    a.append(i)
    print(f"Вероятность наступления дефолта {k} из {n} равна {poisson.pmf(n,p)} {sum(a)}")

Вероятность наступления дефолта 1 из 50 равна 3.628599372152899e-150 0.37160171437460954
Вероятность наступления дефолта 2 из 50 равна 3.628599372152899e-150 0.5574025715619143
Вероятность наступления дефолта 3 из 50 равна 3.628599372152899e-150 0.6180722392149117
Вероятность наступления дефолта 4 из 50 равна 3.628599372152899e-150 0.6326205778868039
Вероятность наступления дефолта 5 из 50 равна 3.628599372152899e-150 0.6353521026986694
Вероятность наступления дефолта 6 из 50 равна 3.628599372152899e-150 0.6357701932310977
Вероятность наступления дефолта 7 из 50 равна 3.628599372152899e-150 0.6358238258358407
Вероятность наступления дефолта 8 из 50 равна 3.628599372152899e-150 0.6358297090042182
Вероятность наступления дефолта 9 из 50 равна 3.628599372152899e-150 0.6358302693059684
Вероятность наступления дефолта 10 из 50 равна 3.628599372152899e-150 0.6358303161883597
Вероятность наступления дефолта 11 из 50 равна 3.628599372152899e-150 0.6358303196675724
Вероятность наступления дефол

In [72]:
ESF1 = 0.636 * 1000 / 0.01
ESF5 = 0.636 * 1000 / 0.05
print(f"ESF1%(X)= {ESF1}, ESF5%(X)= {ESF5}")

ESF1%(X)= 63600.0, ESF5%(X)= 12720.0


Решение задачи 3:

R =1000;

VaR1%(X) =1000;

VaR5%(X) =1000;

ESF1%(X) =50000;

ESF5%(X) =12720.

Задача 4. 

Пусть CF1 = · · · = CF50 = 1000 — одинаковые задолженности от
большого количества (50) контрагентов. 

Пусть ξi — случайная величина,означающая дефолт i-го контрагента. 

Пусть есть общий фактор для всех контрагентов, т.е. ξi = η + ζi, где P(η = 1) = P(ζi = 1) = 0.01,(ζ1, . . . , ζ50)
— независимые случайные величины, P(η = 1, ζi = 1) = 0.

R =?; V aR1%(X) =?; V aR5%(X) =?; ESF1%(X) =?; ESF5%(X) =?.

In [73]:
n=50
p1 = 0.01
b = []
for k in range(1, 51):
    i = binom.pmf(k,n,p1)
    b.append(i)
    print(f"Вероятность наступления дефолта {k} из {n} равна {poisson.pmf(n,p1)} {sum(b)}")

Вероятность наступления дефолта 1 из 50 равна 3.255233773313457e-165 0.3055586197664328
Вероятность наступления дефолта 2 из 50 равна 3.255233773313457e-165 0.38117666203186307
Вероятность наступления дефолта 3 из 50 равна 3.255233773313457e-165 0.3933977597717306
Вероятность наступления дефолта 4 из 50 равна 3.255233773313457e-165 0.39484824359439163
Вероятность наступления дефолта 5 из 50 равна 3.255233773313457e-165 0.3949830360304369
Вероятность наступления дефолта 6 из 50 равна 3.255233773313457e-165 0.39499324757862214
Вероятность наступления дефолта 7 из 50 равна 3.255233773313457e-165 0.39499389593088785
Вероятность наступления дефолта 8 из 50 равна 3.255233773313457e-165 0.39499393113183157
Вероятность наступления дефолта 9 из 50 равна 3.255233773313457e-165 0.3949939327911353
Вероятность наступления дефолта 10 из 50 равна 3.255233773313457e-165 0.39499393285985396
Вероятность наступления дефолта 11 из 50 равна 3.255233773313457e-165 0.39499393286237805
Вероятность наступления

In [74]:
ESF1 = (0.395 * 1000 + 50000 * 0.01)/ 0.01
ESF5 = (0.395 * 1000 + 50000 * 0.01)/ 0.05
print(f"ESF1%(X)= {ESF1}, ESF5%(X)= {ESF5}")

ESF1%(X)= 89500.0, ESF5%(X)= 17900.0


Решение задачи 4:

R =1000;

VaR1%(X) =50000;

VaR5%(X) =1000;

ESF1%(X) =50000;

ESF5%(X) =17900.